File: apps/web/src/types/testScenario.ts
export interface TestScenarioRequirement {
  id: number;
  requirement_code: string;
  module: string;
}

export interface TestScenario {
  id: number;
  scenario_code: string;

  requirement_id: number;
  module: string;

  title: string;
  description: string | null;

  priority: string;
  status: string;

  requirement: TestScenarioRequirement;

  created_at: string;
  updated_at: string;
}

export interface TestScenarioRequest {
  requirement_id: number;
  module: string;

  title: string;
  description: string | null;

  priority: string;
 status: string;
}

File: apps/api/app/schemas/test_scenario.py
from datetime import datetime

from pydantic import BaseModel


class TestScenarioRequirementResponse(BaseModel):
    id: int
    requirement_code: str
    module: str

    class Config:
        from_attributes = True


class TestScenarioBase(BaseModel):
    requirement_id: int
    module: str
    title: str
    description: str | None = None
    priority: str
    status: str


class TestScenarioCreate(TestScenarioBase):
    pass


class TestScenarioUpdate(TestScenarioBase):
    pass


class TestScenarioResponse(TestScenarioBase):
    id: int
    scenario_code: str
    requirement: TestScenarioRequirementResponse
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True